docs/articles/licensing/gplv3/gplv3-blacklist.rst

```diff
@@ -0,0 +1,86 @@
+GPLv3 licensed production software Vs automotive industry
+---------------------------------------------------------
+
+Modern cars have around 100 million lines of code running on them,12 with 70% 
+of that code being in the head unit. Complexity is a non-trivial issue in 
+automotive software design. In addition to being complex, cars can be dangerous.
+
+Locking the software to the hardware by signing the original software image
+with a cryptographic key so that only an image provided by the supplier will 
+boot or install – is a common practice in embedded devices.
+This process of signing software images – so only images with the right key 
+will boot or install effectively prevents a user from modifying the software
+on the device since they have no access to the key needed to allow their 
+modified version to boot or install.
+
+With regard to the GPLv3, the difficulty is with only the anti-tivoization 
+clause, and the reason for disfavoring that license is safety. 
+
+References
+^^^^^^^^^^
+
+`<https://www.gnu.org/licenses/gpl-faq.en.html#Tivoization>`_
+`<http://www.ifosslr.org/ifosslr/article/view/102/207>`_
+
+Difference between GPLv3 license usage of production and development images
+---------------------------------------------------------------------------
+
+* The development image can contain GPLv3 softwares even without commercial
+  license. E.g - gdbserver(GPLv3) can be used in development image 
+  but not in production image.
+* In production image if GPLv3 non-commercial licensed software is used,
+  then the vendor has to provide the necessary tools to change the 
+  software as well.
+
+
+How to facilitate blacklisting of GPLv3 licensed software in Yocto
+------------------------------------------------------------------
+
+1. Add layer **meta-gplv2**
+2. Remove "initramfs-live-install" and "initramfs-module-install" from 
+   environment variable PACKAGE_INSTALL.
+   *(file: core-image-minimal-initramfs.bbappend)*
+
+   .. code-block:: bash
+
+      PACKAGE_INSTALL_remove = "initramfs-live-install initramfs-module-install"
+3. Remove "grub-efi" from WKS_FILE_DEPENDS_BOOTLOADERS for x86-64 arch.
+   *(file: image_types_wic.bbclass)*
+
+   .. code-block:: bash
+
+     WKS_FILE_DEPENDS_BOOTLOADERS_remove_x86-64 = "grub-efi"
+4. Remove "parted" from RDEPENDS_${PN}. 
+   *(file: initramfs-module-install-efi_1.0.bb)*
+
+   .. code-block:: bash
+
+     RDEPENDS_${PN}_remove = "parted"
+5. Select "system-boot" as the bootloader instead of default "grub-efi".
+   *(file: systemd-boot.bbclass)*
+
+   .. code-block:: bash
+
+     EFI_PROVIDER = "systemd-boot"
+6. Deprecate GPLv3 and other unwanted licenses as follows - 
+
+   .. code-block:: bash
+
+	INCOMPATIBLE_LICENSE = "\
+	    GPL-3.0 \
+	    LGPL-3.0 \
+	    AGPL-3.0 \
+	    MPL-1.1 \
+	    SGI-1 \
+	    CECILL-2.0 \
+	    CDDL-1.1 \
+	    EPL-1.0 \
+	    CPL-1.0 \
+	"
+	PACKAGECONFIG_remove = "cairo"
```

Comment:
Why is cairo here in the base image? What component depends on it?

Reply:
The answer to why a given component is in PELUX should really come from the PELUX Architect.

Reply:
@GordanM NTG-X guys have removed "cairo" as part of there GPLv3 licensed software removal activity. That is why they don't get any error on python3-pycairo. What should we do now? Shall I remove 3g DISTRO FEATURE or shall I remove cairo?

Reply:
@GordanM -
1. The test package is part of ofono itself. They are just python files and called during yocto qa API calls. This is a run time dependency.
2.  Tried with "--disable-test" and without any flag as well in CMake flag macro "EXTRA_OECONF", It doesn't help in solving the problem. So CMake doesn't play a role in getting the test package in.
3.  In order to remove python3-pycairo we can remove the "3g" feature as we have done for "bluetooth" in other project.

Reply:
If the components are only for testing, then they are not required on the target and as such should be ignored. Why spend time and energy on something that will never be installed on the target?

Reply:
@jeremiah agreed!